plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/StaticTokenAwareHttpMetastoreClientFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.trino.spi.security.ConnectorIdentity;
+import org.apache.thrift.TException;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class StaticTokenAwareHttpMetastoreClientFactory
+        implements IdentityAwareMetastoreClientFactory
+{
+    private final ThriftMetastoreClientFactory clientProvider;
+    private final ImmutableList<URI> metastoreUris;
+
+    @Inject
+    StaticTokenAwareHttpMetastoreClientFactory(StaticMetastoreConfig config, ThriftMetastoreClientFactory clientProvider)
+    {
+        this.clientProvider = requireNonNull(clientProvider, "clientProvider is null");
+        this.metastoreUris = config.getMetastoreUris().stream()
+                .map(StaticTokenAwareHttpMetastoreClientFactory::checkMetastoreUri)
+                .collect(toImmutableList());
+    }
+
+    private static URI checkMetastoreUri(URI uri)
+    {
+        requireNonNull(uri, "uri is null");
+        String scheme = uri.getScheme();
+        checkArgument(!isNullOrEmpty(scheme), "metastoreUri scheme is missing: %s", uri);
+        checkArgument(scheme.equals("thrift") || scheme.equals("https") || scheme.equals("http"), "metastoreUri scheme must be thrift, http or https: %s", uri);
```

Comment:
Why do we allow "thrift" in  StaticTokenAware **_Http_** MetastoreClientFactory?

Reply:
Good point. Updated the check to check only for http schemes.